perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -297,6 +298,18 @@ public Member createSpecificMember(PerunSession sess, Vo vo, Candidate candidate
 		return member;
 	}
 
+	public Member createSponsoredAccount(PerunSession sess, Map<String, String> params, String namespace, ExtSource extSource, String extSourcePostfix, User owner, Vo vo) throws InternalErrorException, ExtendMembershipException, AlreadyMemberException, WrongReferenceAttributeValueException, WrongAttributeValueException, GroupOperationsException {
+		String loginNamespaceUri = "urn:perun:user:attribute-def:def:login-namespace:" + namespace;
+		Map<String, String> generatedParams = getPerunBl().getUsersManagerBl().generateAccount(sess, namespace, params);
```

Comment:
To be most generic, we should call generateAccount() only when input map ``params`` doesn't contain or has null value for key stored in ``loginNamespaceUri``. Case - caller doesn't provided new login for new user. If caller provided login, then we should call reservePassword() and validatePasswordAndSetUserExtSources(), but it must be called after actual user/member creation and ``password`` param must be present too.